File: examples/typescript/04_sessions_and_memory.ts
// Sessions remember the conversation. Memory remembers facts across conversations.
import { Agent, FileMemory, FileSession, run } from "openharness";

const memory = new FileMemory("data/memory.json"); // durable facts, shared by every session
const agent = new Agent({ instructions: "You are a personal assistant. Save lasting preferences with remember.", memory });

const monday = new FileSession("data/monday.jsonl");
await run(agent, "Hi, I'm Sam. I always want temperatures in Celsius.", { session: monday });
console.log((await run(agent, "What's my name?", { session: monday })).output);

const tuesday = new FileSession("data/tuesday.jsonl"); // new conversation, same memory
console.log((await run(agent, "Which temperature unit do I prefer?", { session: tuesday })).output);
console.log("memories:", (await memory.list()).map((m) => m.text));

File: examples/python/04_sessions_and_memory.py
"""Sessions remember the conversation. Memory remembers facts across conversations."""

import asyncio

from openharness import Agent, FileMemory, SQLiteSession, run

memory = FileMemory("data/memory.json")  # durable facts, shared by every session
agent = Agent(instructions="You are a personal assistant. Save lasting preferences with remember.", memory=memory)


async def main() -> None:
    monday = SQLiteSession("monday", "data/chats.db")
    await run(agent, "Hi, I'm Sam. I always want temperatures in Celsius.", session=monday)
    print((await run(agent, "What's my name?", session=monday)).output)

    tuesday = SQLiteSession("tuesday", "data/chats.db")  # new conversation, same memory
    print((await run(agent, "Which temperature unit do I prefer?", session=tuesday)).output)
    print("memories:", [m.text for m in await memory.list()])


asyncio.run(main())
